src/main/java/io/lighty/yang/validator/formats/Line.java

```diff
@@ -45,70 +43,48 @@
     private static final String ANYDATA = "<anydata>";
 
     final RpcInputOutput inputOutput;
-    final List<Integer> removeChoiceQname;
-    private final Map<XMLNamespace, String> namespacePrefix;
-
     final List<IfFeatureStatement> ifFeatures = new ArrayList<>();
     final List<String> keys = new ArrayList<>();
     final boolean isMandatory;
     final boolean isListOrLeafList;
     final boolean isChoice;
     final boolean isCase;
+    private final Map<XMLNamespace, String> namespacePrefix;
     Status status;
     String nodeName;
     String flag;
     String path;
     String typeName;
 
-    Line(final LyvNodeData lyvNodeData, final RpcInputOutput inputOutput, final List<Integer> removeChoiceQname,
+    Line(final LyvNodeData lyvNodeData, final RpcInputOutput inputOutput,
             final Map<XMLNamespace, String> namespacePrefix) {
-        SchemaNode node = lyvNodeData.getNode();
+        final SchemaNode node = lyvNodeData.getNode();
         this.status = node.getStatus();
         this.isMandatory = lyvNodeData.isNodeMandatory();
         this.isListOrLeafList = node instanceof LeafListSchemaNode || node instanceof ListSchemaNode;
         this.isChoice = node instanceof ChoiceSchemaNode;
         this.isCase = node instanceof CaseSchemaNode;
         this.nodeName = node.getQName().getLocalName();
         this.inputOutput = inputOutput;
-        this.removeChoiceQname = removeChoiceQname;
         this.namespacePrefix = namespacePrefix;
-        resolveFlag(node, lyvNodeData.getContext());
+        resolveFlag(node, lyvNodeData.getAbsolutePath(), lyvNodeData.getContext());
         resolvePathAndType(node);
         resolveKeys(node);
         resolveIfFeatures(node);
     }
 
-    protected abstract void resolveFlag(SchemaNode node, SchemaContext context);
+    protected abstract void resolveFlag(SchemaNode node, Absolute absolutePath, EffectiveModelContext context);
 
-    protected void resolveFlagForDataSchemaNode(final SchemaNode node, final SchemaContext context,
```

Comment:
We do not solve isConfig problem this in this PR.

Reply:
Sorry don't understand, Here is removed workaround with choice nodes. Choice nodes should be removed from path before calling `context.findDataTreeChild(qNames)`. This was resolved with SchemaIS.

Reply:
revert this back: `if (dataSchemaNode.effectiveConfig().orElse(Boolean.TRUE)) {`

Reply:
Done